// @giphy/colors/dist/index.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  deleteColor: () => deleteColor,
  dimColor: () => dimColor,
  errorColor: () => errorColor,
  facebookColor: () => facebookColor,
  gifOverlayColor: () => gifOverlayColor,
  giphyAqua: () => giphyAqua,
  giphyBlack: () => giphyBlack,
  giphyBlue: () => giphyBlue,
  giphyCharcoal: () => giphyCharcoal,
  giphyDarkCharcoal: () => giphyDarkCharcoal,
  giphyDarkGrey: () => giphyDarkGrey,
  giphyDarkestGrey: () => giphyDarkestGrey,
  giphyGreen: () => giphyGreen,
  giphyIndigo: () => giphyIndigo,
  giphyLightBlue: () => giphyLightBlue,
  giphyLightCharcoal: () => giphyLightCharcoal,
  giphyLightGrey: () => giphyLightGrey,
  giphyLightestGrey: () => giphyLightestGrey,
  giphyPink: () => giphyPink,
  giphyPurple: () => giphyPurple,
  giphyRed: () => giphyRed,
  giphyWhite: () => giphyWhite,
  giphyWhiteSmoke: () => giphyWhiteSmoke,
  giphyYellow: () => giphyYellow,
  instagramColor: () => instagramColor,
  pinterestColor: () => pinterestColor,
  primaryCTA: () => primaryCTA,
  primaryCTADisabled: () => primaryCTADisabled,
  redditColor: () => redditColor,
  secondaryCTA: () => secondaryCTA,
  smsColor: () => smsColor,
  tumblrColor: () => tumblrColor,
  twitterColor: () => twitterColor
});
module.exports = __toCommonJS(src_exports);
var giphyBlack = "#121212";
var giphyDarkestGrey = "#212121";
var giphyDarkGrey = "#2e2e2e";
var giphyDarkCharcoal = "#3e3e3e";
var giphyCharcoal = "#4a4a4a";
var giphyLightCharcoal = "#5c5c5c";
var giphyLightGrey = "#a6a6a6";
var giphyLightestGrey = "#d8d8d8";
var giphyWhiteSmoke = "#ececec";
var giphyWhite = "#ffffff";
var giphyBlue = "#00ccff";
var giphyGreen = "#00ff99";
var giphyPurple = "#9933ff";
var giphyRed = "#ff6666";
var giphyYellow = "#fff35c";
var giphyAqua = "#00e6cc";
var giphyLightBlue = "#3191ff";
var giphyIndigo = "#6157ff";
var giphyPink = "#e646b6";
var facebookColor = "#3894fc";
var twitterColor = "#00ccff";
var pinterestColor = "#e54cb5";
var tumblrColor = "#529ecc";
var instagramColor = "#c23c8d";
var redditColor = "#fc6669";
var smsColor = "#00ff99";
var errorColor = giphyRed;
var deleteColor = giphyRed;
var primaryCTA = giphyIndigo;
var primaryCTADisabled = "#241F74";
var secondaryCTA = giphyCharcoal;
var dimColor = `rgba(0, 0, 0, 0.8)`;
var gifOverlayColor = `rgba(0, 0, 0, 0.4)`;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  deleteColor,
  dimColor,
  errorColor,
  facebookColor,
  gifOverlayColor,
  giphyAqua,
  giphyBlack,
  giphyBlue,
  giphyCharcoal,
  giphyDarkCharcoal,
  giphyDarkGrey,
  giphyDarkestGrey,
  giphyGreen,
  giphyIndigo,
  giphyLightBlue,
  giphyLightCharcoal,
  giphyLightGrey,
  giphyLightestGrey,
  giphyPink,
  giphyPurple,
  giphyRed,
  giphyWhite,
  giphyWhiteSmoke,
  giphyYellow,
  instagramColor,
  pinterestColor,
  primaryCTA,
  primaryCTADisabled,
  redditColor,
  secondaryCTA,
  smsColor,
  tumblrColor,
  twitterColor
});
